### PR TITLE
Avoid blocking calls in test

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
Not sure if it makes a difference, but why not?

File: src/ProjectTemplates/test/Helpers/ProcessEx.cs [hunk 1]
```diff
@@ -170,20 +171,16 @@ internal string GetFormattedOutput()
...
             }
 
-            var exited = Exited.Wait(timeSpan.Value);
-            if (!exited)
-            {
-                _output.WriteLine($"The process didn't exit within the allotted time ({timeSpan.Value.TotalSeconds} seconds).");
-                _process.Dispose();
-            }
-            else if (assertSuccess && _process.ExitCode != 0)
+            await Exited.TimeoutAfter(timeSpan.Value);
```
Comment (ajaybhargavb) [inline src/ProjectTemplates/test/Helpers/ProcessEx.cs]: Does it blow up or gracefully exit if there is a timeout? If it exits silently, will the TimeOut be logged in the StdOut?

Comment (pranavkm) [reply src/ProjectTemplates/test/Helpers/ProcessEx.cs]: It throws: https://github.com/aspnet/Extensions/blob/24c84a5b1db0fca3cfcc54eed3f4f94c6670a709/src/TestingUtils/Microsoft.AspNetCore.Testing/src/TaskExtensions.cs#L33. It's a slightly different error message, but still useful.

Comment (ajaybhargavb) [reply src/ProjectTemplates/test/Helpers/ProcessEx.cs]: Interesting. This is different from what was before. Are you sure about letting it blow up for every timeout? Might be worth just logging it and be hopeful that the test still passes. There are a bunch of intermittent flakiness here that could cause random timeouts that may not affect the test.

**Review by javiercn — APPROVED** [review]
I don't have strong feeligns about this

**Pull request closed without merging**